runtime/gc_glue_java/ObjectModelDelegate.cpp

```diff
@@ -84,3 +84,24 @@ GC_ObjectModelDelegate::calculateObjectDetailsForCopy(MM_EnvironmentBase *env, M
 	*hotFieldAlignmentDescriptor = clazz->instanceHotFieldDescription;
 }
 #endif /* defined(OMR_GC_MODRON_SCAVENGER) */
+
+void
+GC_ObjectModelDelegate::initializeMinimumSizeObject(MM_EnvironmentBase *env, void *allocAddr)
+{
+	J9JavaVM *javaVM = (J9JavaVM *)env->getLanguageVM();
+	J9Class *clazz = J9VMJAVALANGOBJECT_OR_NULL(javaVM);
+	j9object_t instance = (j9object_t) allocAddr;
+
+	memset(instance, 0, OMR_MINIMUM_OBJECT_SIZE);
+
+	if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(javaVM)) {
+		J9ObjectCompressed *objectHeader = (J9ObjectCompressed*) instance;
+		objectHeader->clazz = (uint32_t)(uintptr_t)clazz;
+	} else {
+		J9ObjectFull *objectHeader = (J9ObjectFull*) instance;
+		objectHeader->clazz = (uintptr_t)clazz;
+	}
```

Comment:
try this: MM_GCExtensions::getExtensions(env)->objectModel.setObjectClass(instance, clazz);

Reply:
Pushed the change, testing it now